lib/checkPullRequestReview.js

```diff
@@ -0,0 +1,290 @@
+ * 5. If reviewer was the last reviewer, check if pull request has been
+ *    approved, and add the LGTM Label.
+ * 6. If pull request has been approved, assign PR author or last reviewer
+ *    to merge the PR.
+ *
+ * @param {import('probot').Context} context
+ * @param {import('probot').Octokit.PullsGetResponse} pullRequest
+ */
+const handleApproval = async (context, pullRequest) => {
+  /**
+   * @type {import('probot').Octokit.PullsGetReviewResponse} review
+   */
+  const review = context.payload.review;
+  const reviewer = review.user.login;
+
+  const reviewerIsAssigned = pullRequest.assignees.some(
+    (assignee) => assignee.login === reviewer
+  );
+
+  if (reviewerIsAssigned) {
+    // Unassign the reviewer if still assigned.
+    await context.github.issues.removeAssignees(
+      context.repo({
+        issue_number: pullRequest.number,
+        assignees: [reviewer],
+      })
+    );
+
+    await context.github.issues.createComment(
+      context.repo({
+        issue_number: pullRequest.number,
+        body: 'Unassigning @' + reviewer + ' since the PR is approved.',
+      })
+    );
+  }
+
+  // Check if pull request has been approved by all reviewers.
+  // The pull request data does not contain information if the pull request
+  // has been approved, so we are searching for an approved pull request
+  //  with the pull request number and in the appropriate repository. An
+  // empty array will be returned if the pull request is not approved by
+  // all reviewers.
+  const searchResult = await context.github.search.issuesAndPullRequests(
+    context.repo({
+      q:
+        'repo:' +
+        context.payload.repository.full_name +
+        ' review:approved ' +
+        pullRequest.number,
+    })
+  );
+
+  const searchData = searchResult.data.items;
+  const assignees = pullRequest.assignees.map((assignee) => assignee.login);
+  if (searchData.length === 0) {
+    // Pull request has not been approved by all reviewers, hence we need to
+    // assign the remaining reviewers.
+    const reviewersYetToReview = pullRequest.requested_reviewers.map(
+      (reviewer) => reviewer.login
+    );
+
+    const unassignedReviewers = reviewersYetToReview.filter(
+      (reviewer) => !assignees.includes(reviewer)
+    );
+    if (unassignedReviewers.length > 0) {
+      // Ping and assign unassigned reviewers.
+      const commentBody =
+        'Assigning @' +
+        unassignedReviewers.join(', @') +
+        ' for code owner reviews, Thanks!';
+
+      await context.github.issues.createComment(
+        context.repo({
+          issue_number: pullRequest.number,
+          body: commentBody,
+        })
+      );
+
+      await context.github.issues.addAssignees(
+        context.repo({
+          issue_number: pullRequest.number,
+          assignees: unassignedReviewers,
+        })
+      );
+    }
+  } else {
+    // The pull request has been approved, and we need to add the LGTM label.
```

Comment:
Since the pull request has been approved, we add the LGTM label.

Reply:
Done.